expandedTree/expandedTree.go

```diff
@@ -26,12 +26,13 @@ type ETree struct {
 }
 
 //read tree from filename and add fastas and up pointers to the tree
-func ReadTree(newickFilename string, fastasFilename string) *ETree {
+func ReadTree(newickFilename string, fastasFilename string) (*ETree, error) {
 	tr, err := ReadNewick(newickFilename)
-	if err == nil {
+	if err != nil {
+		return nil, err
 	}
 	AssignFastas(tr, fastasFilename)
-	return tr
+	return tr, nil
 }
 
 //read in tree from filename
```

Comment:
I've been trying to wrap my head around some of the code in expandedTree.go but it is old code so I can't necessarily point to it. I will leave my questions here and try to be explicit about what I'm talking about. 
1. For the ReadNewick() ---    Can someone explain in words what is happening at newline 50?
2. "        "  ---- for newline 53, will this only run if the two above "if" statements fail? (newlines 44 and 49) I'm expecting an "else" at this line, but without it my previous understanding if that the "if" statements would be checked, and once completed this "return" statement would always run. 
2. I'm trying to work through all the possible error statements that can come through the edited ReadTree() in expandedTree.go. 
      a. I see an error from the ReadNewick()
             i. This error could come from the os.Open()
            ii. This error could come from the "return nil, errors.New(..." defined on newline 53. 
            iii. Could this also have an error fed in from the "if !strings.HasPrefix()....." which would essentially come from the parseNewick()?
                 1. If yes, then an error can be from the parseNewick() statement if the line (tree) fails to start with "(" and end with ";"
                 2. Also, an error could come from the parseNewickHelper()?
                       i. For errors in this function..... I was reading about Newick tree format and code at newline 165 would fail for the tree I'm reading about? https://evolution.genetics.washington.edu/phylip/newicktree.html 
                           - does this code at 165 work for all trees? or just a very specific example tree?
                       ii. I see all the possible errors in the first batch of if/else statement. 
                      iii. we can get errors from the splitNameAndLength() which I believe has thwo possible error scenarios within itself.  
                      iv. But I do not understand newlines 185-192?